server/src/versioning.rs

```diff
@@ -0,0 +1,99 @@
+use iggy::error::IggyError;
+use std::fmt::Display;
+use std::str::FromStr;
+
+const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+#[derive(Debug)]
+pub struct SemanticVersion {
+    pub major: u32,
+    pub minor: u32,
+    pub patch: u32,
+}
+
+impl FromStr for SemanticVersion {
+    type Err = IggyError;
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        let mut version = s.split('.');
+        let major = version.next().unwrap().parse::<u32>()?;
+        let minor = version.next().unwrap().parse::<u32>()?;
+        let patch = version.next().unwrap().parse::<u32>()?;
+        Ok(SemanticVersion {
+            major,
+            minor,
+            patch,
+        })
+    }
+}
+
+impl SemanticVersion {
```

Comment:
why did you implement it manually instead of using semver crate?

Reply:
We've had it for a long time now as it was very easy to (de)serialize to/from bytes, we could switch to the actual library if needed :)